### PR TITLE
bump python, numpy

### LINKED ISSUE #194: Python/NumPy support matrix
It would be good to support the same Python/NumPy as CIL

https://github.com/TomographicImaging/CIL/issues/1719

Python:
  - 3.10
  - 3.11
  - 3.12

NumPy:
  - 1.23
  - 1.24
  - 1.25
  - 1.26

### PR DESCRIPTION
**Pull request by casperdcl**
bump python & numpy versions

- conda (fixes #194)
- CI matrix

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -52,10 +52,10 @@ jobs:
     strategy:
       matrix:
         include:
-        - python-version: 3.8
-          numpy-version: 1.21
         - python-version: '3.10'
-          numpy-version: 1.24
+          numpy-version: 1.23
+        - python-version: 3.12
+          numpy-version: 1.26
```
Comment (gfardell) [inline .github/workflows/build.yml:58]: Is this testing the minimum and the max version combos?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -74,8 +74,8 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.9]
-        numpy-version: [1.22]
+        python-version: [3.11]
+        numpy-version: [1.25]
```
Comment (gfardell) [inline .github/workflows/build.yml:78]: What are these versions for?

**Review by gfardell — APPROVED** [review]
More queries for me than PR comments. What's the logic in the GHA versions?

File: .github/workflows/build.yml (hunk 1 above)
Comment (casperdcl) [reply .github/workflows/build.yml:58]: Yup

File: .github/workflows/build.yml (hunk 2 above)
Comment (casperdcl) [reply .github/workflows/build.yml:78]: penultimate version. Just being safe as in my experience GPU stuff is temperamental on latest versions.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -52,10 +52,10 @@ jobs:
     strategy:
       matrix:
         include:
```
Comment (casperdcl) [inline .github/workflows/build.yml:54]:
```suggestion
         # min & max versions
         include:
```

**Pull request merged**